src/libFLAC/CMakeLists.txt

```diff
@@ -10,14 +10,29 @@ check_include_file("sys/param.h" HAVE_SYS_PARAM_H)
 set(CMAKE_REQUIRED_LIBRARIES m)
 check_function_exists(lround HAVE_LROUND)
 
-if(CMAKE_SYSTEM_PROCESSOR MATCHES "[xX]86_64|(AMD|amd)64")
-    if(CMAKE_SIZEOF_VOID_P EQUAL 4)
-        set(IA32 TRUE)
-    endif()
-    add_definitions(-DFLAC__CPU_X86_64 -DFLAC__ALIGN_MALLOC_DATA)
-elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "([xX]|i[346])86")
```

Comment:
imo this regex could be improved:
"[xX]86|i.86"
would catch the currently missing i586 architecture